rust/main/chains/hyperlane-sealevel/src/tx_submitter.rs

```diff
@@ -0,0 +1,121 @@
+#![allow(dead_code)]
+
+use async_trait::async_trait;
+use hyperlane_core::ChainResult;
+use solana_sdk::{
+    compute_budget::ComputeBudgetInstruction, instruction::Instruction, pubkey::Pubkey,
+    signature::Signature, transaction::Transaction,
+};
+
+use crate::SealevelRpcClient;
+
+/// A trait for submitting transactions to the chain.
+#[async_trait]
+pub trait TransactionSubmitter: Send + Sync {
+    /// Get the instruction to set the compute unit price.
+    fn get_priority_fee_instruction(
+        &self,
+        compute_unit_price_micro_lamports: u64,
+        compute_units: u64,
+        payer: &Pubkey,
+    ) -> Instruction;
+
+    /// Send a transaction to the chain.
+    async fn send_transaction(
+        &self,
+        transaction: &Transaction,
+        skip_preflight: bool,
+    ) -> ChainResult<Signature>;
+}
+
+/// A transaction submitter that uses the vanilla RPC to submit transactions.
+#[derive(Debug)]
+pub struct RpcTransactionSubmitter {
+    rpc_client: SealevelRpcClient,
+}
+
+impl RpcTransactionSubmitter {
+    pub fn new(url: String) -> Self {
+        Self {
+            rpc_client: SealevelRpcClient::new(url),
+        }
+    }
+
+    pub fn new_with_rpc_client(rpc_client: SealevelRpcClient) -> Self {
+        Self { rpc_client }
+    }
+}
+
+#[async_trait]
+impl TransactionSubmitter for RpcTransactionSubmitter {
+    fn get_priority_fee_instruction(
+        &self,
+        compute_unit_price_micro_lamports: u64,
+        _compute_units: u64,
+        _payer: &Pubkey,
+    ) -> Instruction {
+        ComputeBudgetInstruction::set_compute_unit_price(compute_unit_price_micro_lamports)
+    }
+
+    async fn send_transaction(
+        &self,
+        transaction: &Transaction,
+        skip_preflight: bool,
+    ) -> ChainResult<Signature> {
+        self.rpc_client
+            .send_transaction(transaction, skip_preflight)
```

Comment:
when you say this approach uses priority fee auctions in the PR description, is that implicit to this call?

Reply:
yep exactly